mlir-tensorrt/test/Conversion/TensorRTRuntimeToExecutor/tensorrt-runtime-to-executor.mlir

```diff
@@ -133,4 +133,93 @@ func.func @main(%arg0: memref<1x3x256x256xf32, #executor.memory_type<device>>) -
+// CHECK: }
+// CHECK: executor.constant_resource @my_func_engine_data dense<0> : vector<10xi8>
+// CHECK-LABEL: executor.global @my_func_exec_ctx constant : !executor.opaque<"trtrt_context"> {
+// CHECK-DAG: %[[v1:.*]] = executor.load_constant_resource @my_func_engine_data : !executor.ptr<host>
+// CHECK-DAG: %[[v2:.*]] = executor.getoffset[10] : () -> i64, i8
+// CHECK-DAG: %[[v3:.*]] = executor.get_global @tensorrt_runtime : !executor.opaque<"trtrt_runtime">
+// CHECK: %[[v4:.*]] = executor.call @_trtrt_load(%[[v3]], %[[v1]], %[[v2]]) : (!executor.opaque<"trtrt_runtime">, !executor.ptr<host>, i64) -> !executor.opaque<"trtrt_engine">
+// CHECK: %[[v5:.*]] = executor.call @_trtrt_create_context(%[[v4]]) : (!executor.opaque<"trtrt_engine">) -> !executor.opaque<"trtrt_context">
+// CHECK: executor.return %[[v5]] : !executor.opaque<"trtrt_context">
+// CHECK: }
+// CHECK-LABEL: tensorrt.module @trt_funcs {
+// CHECK: func.func @my_func() attributes {tensorrt.engine = dense<0> : vector<10xi8>}
+// CHECK: }
+// CHECK-LABEL: func.func @main
+// CHECK-SAME: (%[[arg0:.*]]: memref<?x?x?xf32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 1, 2], max = [1, 4, 2]>}, %[[arg1:.*]]: memref<?x?xi32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 1], max = [1, 4]>}) -> (memref<?x?x256xf32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 2, 256], max = [1, 5, 256]>}, memref<?x256x64x64xf32, #executor.memory_type<device>> {tensorrt.shape_profile = #executor.dim_bounds<min = [1, 256, 64, 64], max = [1, 256, 64, 64]>})
+// CHECK-DAG: %[[c4:.*]] = executor.constant 4 : i64
+// CHECK-DAG: %[[c3:.*]] = executor.constant 3 : i64
+// CHECK-DAG: %[[c2:.*]] = executor.constant 2 : i64
+// CHECK-DAG: %[[c0:.*]] = executor.constant 0 : i64
+// CHECK-DAG: %[[c1:.*]] = executor.constant 1 : i64
+// CHECK: %[[v6:.*]] = builtin.unrealized_conversion_cast %[[arg0]] : memref<?x?x?xf32, #executor.memory_type<device>> to !executor.table<!executor.ptr<device>, !executor.ptr<device>, i64, i64, i64, i64, i64, i64, i64>
+// CHECK: %[[v7:.*]] = builtin.unrealized_conversion_cast %[[arg1]] : memref<?x?xi32, #executor.memory_type<device>> to !executor.table<!executor.ptr<device>, !executor.ptr<device>, i64, i64, i64, i64, i64>
+// CHECK: %[[v8:.*]] = executor.get_global @my_func_exec_ctx : !executor.opaque<"trtrt_context">
+// CHECK: %[[v9:.*]] = cuda.get_global_stream 0
+// CHECK: %[[v10:.*]] = builtin.unrealized_conversion_cast %[[v9]] : !cuda.stream to !executor.ptr<host>
+// CHECK: %[[v11:.*]] = executor.alloca %[[c1]] x !executor.table<i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64> : (i64) -> !executor.ptr<host>
+// CHECK: %[[v12:.*]] = executor.getoffset[0, 0] : () -> i64, !executor.table<i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64>
```

Comment:
Nit: for these sorts of CHECK statements in very verbose IR, I tend to try to knock off as much extraneous detail as possible that doesn't really add anything to the test quality. Try to do a simple regex search/replace that eliminates everything after the `:` in all these check statements. That removes a lot of textual detail that isn't really important to check here and makes the test easier to maintain.

Reply:
Make sense.